app/infrastructure/s3/awsclient/JavaAWSClientBasedS3PostSigner.java

```diff
@@ -0,0 +1,150 @@
+package infrastructure.s3.awsclient;
```

Comment:
I'm not reviewing this class thoroughly, it looks like it may work (or not). Would it make sense to have some unit tests for it?

Reply:
Well... :)